util/execdetails/execdetails.go

```diff
@@ -52,3 +54,58 @@ func (d ExecDetails) String() string {
 	}
 	return strings.Join(parts, " ")
 }
+
+// RuntimeStats collects executors's execution info.
+type RuntimeStats map[string]*RuntimeStat
+
+// RuntimeStat collects one executor's execution info.
+type RuntimeStat struct {
+	// executor's Next() called times.
+	loop int32
+	// executor consume time.
+	consume int64
+	// executor return row count.
+	rows int64
+}
+
+// NewRuntimeStats creates new executor collector.
+func NewRuntimeStats() RuntimeStats {
+	return RuntimeStats(make(map[string]*RuntimeStat))
+}
+
+// GetRuntimeStat gets execStat for a executor.
+func (e RuntimeStats) GetRuntimeStat(planID string) *RuntimeStat {
```

Comment:
maybe we need a mutex to prevent data race on the `map`, considered the parallel executed inner children of the index join.

Reply:
we have no way to pre-init inner plainID's entry to make it lock free?

Reply:
ok, GetRuntimeStat is low frequence operation...let's lock.

Reply:
I think a lock is much more easier.